magento/graphql-ce
# Fix for getting shipping/billing address data of any other customer

### Description (*)
This PR provides a fix for an issue when a customer is able to use address ID of another customer for setting billing/shipping address.

### Fixed Issues (if relevant)
1. https://github.com/magento/graphql-ce/issues/375: Customer can get shipping/billing address data of any other customer

### Manual testing scenarios (*)
Please, check preconditions described in the issue.
For checking a shipping address use the following query:
```
mutation {
  setShippingAddressesOnCart(
    input: {
      cart_id: "$maskedQuoteId"
      shipping_addresses: [
        {
          customer_address_id: 1 <!-- Address id of first customer
        }
      ]
    }
  ) {
    cart {
      shipping_addresses {
        firstname
        lastname
        company
        street
        city
        postcode
        telephone
      }
    }
  }
}
```
For checking billing address use the following query:
```
mutation {
  setBillingAddressOnCart(
    input: {
      cart_id: "$maskedQuoteId"
      billing_address: 
        {
          customer_address_id: 1  <!-- Address id of first customer
        }
    }
  ) {
    cart {
      billing_address {
        firstname
        lastname
        company
        street
        city
        postcode
        telephone
      }
    }
  }
}
```
